### PR TITLE
Re-organize and update project

### PR DESCRIPTION
**Pull request by pbrisbin**
Removes the `persistent-arbitrary` package, because we've been creating that instance client side ourselves, and can do so in our test suite too. At that point, the `graphula-core` sub-package made little sense, so I moved all the code to top-level. Then I updated the Stackage resolver, added our usual `-Weverything`, and addressed all the warnings.

### CODE REVIEW

File: src/Graphula/Internal.hs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
```
Comment (pbrisbin) [inline src/Graphula/Internal.hs:11]: I'll address this before merging.

File: test/README.lhs [hunk 2]
```diff
@@ -19,15 +19,17 @@ Graphula is a simple interface for generating persistent data and linking its de
 {-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE UndecidableInstances #-}
-module Main where
+{-# OPTIONS_GHC -fno-warn-missing-deriving-strategies #-}
```
Comment (pbrisbin) [inline test/README.lhs:22]: Necessary for the persistent quasi-quoter.

File: src/Graphula.hs [hunk 3]
```diff
@@ -43,6 +25,23 @@
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE UndecidableSuperClasses #-}
 
+-- |
+--
+-- Graphula is a compact interface for generating data and linking its
+-- dependencies. You can use this interface to generate fixtures for automated
+-- testing.
+--
+-- The interface is extensible and supports pluggable front-ends.
```
Comment (eborden) [inline src/Graphula.hs:34]: Is the pluggable frontends still true?

**Review by eborden — APPROVED** [review]

**Review by cdparks — APPROVED** [review]

**Comment by eborden** [comment]
Looks like you merged this to `master` instead of `main`.

**Comment by pbrisbin** [comment]
Doh. What can we do to make that error and not just create the branch?

**Comment by eborden** [comment]
Right, gotta delete the branch.

**Pull request merged**